### PR TITLE
Added a matomo service

### PR DESCRIPTION
**Pull request by TheSlimvReal**
see issue: #679 

### Visible/Frontend Changes
--

### Architectural/Backend Changes
- Added config for matomo url
- Setting up matomo only if config is set

### CODE REVIEW

File: src/app/core/analytics/matomo.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * This class dynamically adds sets up everything for Matomo.
+ *
+ * The code is inspired by:
+ * https://github.com/Arnaud73/ngx-matomo/blob/master/projects/ngx-matomo/src/lib/matomo-injector.service.ts
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class MatomoService {
```
Comment (sleidig) [inline src/app/core/analytics/matomo.service.ts:12]: I think this `MatomoService` should be part of the existing `AnalyticsService`, whose purpose is to manage everything related to usage analytics (in our case with Matomo as the backend)

**Review by sleidig — APPROVED** [review]
works as intended, thanks! :+1: 

The warning "Piwik not found" seems almost unavoidable because we can only set up the matomo script when AppConfig is loaded - which is after Module imports. Maybe we can optimize this after working on #595 . Tracking is working, however.

File: src/app/core/analytics/analytics.service.ts [hunk 2]
```diff
@@ -7,74 +7,128 @@ import { LoginState } from "../session/session-states/login-state.enum";
 
 const md5 = require("md5");
 
+/**
+ * Track usage analytics data and report it to a backend server like Matomo.
+ *
+ * This is automatically disable if the config.json does not specify "usage_analytics" settings.
```
Comment (TheSlimvReal) [inline src/app/core/analytics/analytics.service.ts:13]:
```suggestion
 * This is automatically disabled if the config.json does not specify "usage_analytics" settings.
```

**Review by TheSlimvReal — COMMENTED** [review]
Looks very good now, with a proper startup process the error should also disapear

**Pull request merged**